modules/dem_uart/test/test_dem_uart.py

```diff
@@ -156,3 +84,139 @@ def test_dem_uart_activation(dut):
     dut._log.info("Check contents of MOD_CS")
     yield access.assert_reg_value(MODULE_DI_ADDRESS, SENDER_DI_ADDRESS,
                                   DiPacket.BASE_REG.MOD_CS.value, 16, 1)
+
+# Mirroring expected transfer through module
+_fifo = []
+
+@cocotb.test()
+def test_dem_to_dii(dut):
+    """
+    Verify that no chars are lost when the DII is busy
+    """
+
+    yield _init_dut(dut)
+    yield _activate_module(dut)
+
+    write_thread = cocotb.fork(dem_to_dii_tx(dut, num_transfers=1000, random_data=True))
+    read_thread  = cocotb.fork(dem_to_dii_rx(dut, num_transfers=1000))
+    
+    # read_thread implicitly waits for write thread to finish as well
+    yield read_thread.join()
+
+@cocotb.coroutine
+def dem_to_dii_tx(dut, num_transfers=500, max_delay=5, random_data=False):
+
+    for i in range(num_transfers):
+
+        delay = random.randint(0, max_delay)
+
+        for _ in range(delay):
+            yield RisingEdge(dut.clk)
+
+        data = (random.randint(0, 255) if random_data else 0x42)
+
+        _fifo.append(data)
+
+        dut.out_valid <= 0x1
+        dut.out_char <= int(data)
```

Comment:
You don't need the 'int()' here since the variable already is an integer (also some other occasions).